classes/s6rc.oeclass

```diff
@@ -8,6 +8,28 @@ RDEPENDS_${PN}:>USE_s6rc = " ${RDEPENDS_S6RC}"
 
 s6rcsrcdir ?= "${sysconfdir}/rc"
 
+addhook s6_use_flags to post_recipe_parse first before set_useflags
+def s6_use_flags(d):
+    oneshot_services = (d.get('S6RC_ONESHOT_SERVICES') or '').split()
+    longrun_services = (d.get('S6RC_LONGRUN_SERVICES') or '').split()
+    bundle_services = (d.get('S6RC_BUNDLE_SERVICES') or '').split()
+
+    recipe_flags = (d.get('RECIPE_FLAGS' or '')).split()
+
+    for sv in oneshot_services + longrun_services:
+        for sfx in ["timeout_up", "timeout_down", "dependencies"]:
+            flag = "USE_" + sv + "s6rc_" + sfx
+            if flag not in recipe_flags:
+                recipe_flags.append(flag)
```

Comment:
I think we need to append to `CLASS_FLAGS` instead of `RECIPE_FLAGS`.

The reason for having both variables is that one is exclusively owned by recipe and the other exclusively owned by classes.

I think we might run into trouble with overrides not doing what is expected otherwise.

For example:

```
S6RC_ONESHOT_SERVICES = "foobar"
RECIPE_FLAGS = "foo"
RECIPE_FLAGS:MACHINE_bar = "bar"
```
which will not see the automatic USE flags for bar machine configuration.